test/uk/gov/hmrc/mtdsatestsupportapi/controllers/requestParsers/DeleteTestBusinessRequestParserSpec.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.mtdsatestsupportapi.controllers.requestParsers
+
+import api.models.domain.Nino
+import api.models.errors.{BadRequestError, BusinessIdFormatError, ErrorWrapper, NinoFormatError}
+import support.UnitSpec
+import uk.gov.hmrc.mtdsatestsupportapi.mocks.validators.MockDeleteTestBusinessValidator
+import uk.gov.hmrc.mtdsatestsupportapi.models.request.deleteTestBusiness.{DeleteTestBusinessRawData, DeleteTestBusinessRequest}
+
+class DeleteTestBusinessRequestParserSpec extends UnitSpec with MockDeleteTestBusinessValidator {
+
+  val parser = new DeleteTestBusinessRequestParser(mockDeleteTestBusinessValidator)
+
+  implicit val correlationId: String = "X-123"
+  val vendorId                       = "some_vendor_id"
+  val validNino                      = "AA999999A"
+  val validBusinessId                = "XAIS12345678910"
```

Comment:
Don't forget the validator is totally mocked in this test. You don't need to call these `validNino`, `validBusinessId` and then use "not_a_valid_nino" etc below in the error cases. You could just call them `nino` and `businessId` and use them everywhere. (It's ok but it does give the impression that the actual values matter, and that the validator is using the values.)